lib/note-toolbar-container.ts

```diff
@@ -129,7 +129,7 @@ const mapDispatchToProps = dispatch => ({
   deleteNoteForever: args => dispatch(deleteNoteForever(args)),
   noteRevisions: args => dispatch(noteRevisions(args)),
   restoreNote: args => dispatch(restoreNote(args)),
-  setEditorMode: args => dispatch(setEditorMode(args)),
+  setEditorMode: mode => dispatch(setEditorMode(mode)),
```

Comment:
No need to do all the work in this PR, but these actions somewhat subvert our normal Redux pattern. All we're doing is passing them to the child component, so how about moving them properly while we're fixing the other aspect of this one Redux state type.

That is, can we eliminate this action from the `<NoteToolBar />` container and instead put it in `<NoteToolbar />` where it's used?

In that file it looks like we're using state-based logic as well when all we care about doing is toggling. If we wanted, we could further remove the duplicated logic and create a second new action-type and let the reducer centralize and isolate that logic…

```ts
// action-types
export type SetEditorMode = Action<'SET_EDITOR_MODE', { mode: T.EditorMode }>;
export type ToggleEditorMode = Action<'TOGGLE_EDITOR_MODE'>;

// reducer
const editorMode: A.Reducer<T.EditorMode> = (state = 'edit', action) => {
   switch (action.type) {
     case 'TOGGLE_EDITOR_MODE':
       return state === 'edit' ? 'markdown' : 'edit';
     case 'SET_EDITOR_MODE':
     case 'App.setEditorMode':
       return action.mode;
     case 'App.newNote':
       return 'edit';
     default:
       return state;
   }
 };
```